compiler/qsc_parse/src/lib.rs

```diff
@@ -108,10 +115,38 @@ impl<T, F: FnMut(&mut ParserContext) -> Result<T>> Parser<T> for F {}
 #[must_use]
 pub fn namespaces(
     input: &str,
+    file_name: Option<&str>,
```

Comment:
I think you really need some test cases where you pass in a value here and validate the AST (for both valid and invalid file names)